### ISSUE TITLE
[Discussion] Interactive mode improvements

### ISSUE THREAD

**Issue by geraldus** [haskell/haskell-mode#1176]
I want to discuss few things about interactive haskell mode.

First, would it make sense to load opened file in REPL immediately when opening very first file in project, that is, if there is no session for visited file yet invoke `haskell-process-load-file`.  Currently, doc-mode shows only keyword hints when no process session is available; also REPL completions is not available until there is no session.

**Comment by gracjan** [comment]
I find myself editing Haskell files as text. Loading them would require a lot of context and setup, so I skip that.

I find it useful to edit files that aren't loaded. Maybe when HIE is ready I'll change my mind.

**Comment by geraldus** [comment]
Hm… I'll try to play with auto-loading files in my local setup when I'll find a time for this; and if I'll have something reasonable I will post here the results.

**Comment by gracjan** [comment]
In #1180 you have another guy that does not want to start sessions and load files immediately.

**Comment by geraldus** [comment]
Well, I start session basically to have nice completions, this speeds up my workflow significantly, in past I used REPL to play with code rarely.  This is because of two things: I wan't used to REPL interactions in general, and when I got used to, there were some notable difference in performance when Emacs REPL outputs a lot of stuff comparing to REPL running in terminal.  But my recent tests show that Emacs' REPL now is quite fast (this may relate to Emacs 25 but I'm not certain), so recently I use REPL much frequently.

Now I also get used to see Eldoc messages and find them quite handy, but this requires active session present.  I dream about Eldoc will show signatures for all things under cursor and for active region using `:type-at`, but this requires collect feature in GHCi (and some hacks for `stack`).

To have up to date information I need to run `haskell-process-load` every time, this is quite simple but requires a habit to use <kbd>C-c C-l</kbd> or (<kbd>C-c C-c</kbd>) instead of <kbd>C-x C-s</kbd> which I don't have yet :D Maybe having some customizations for auto load / reload files would help me and other people too.

Sorry, I'm too verbose as usual.
